service/ec2/customizations.go

```diff
@@ -26,8 +26,12 @@ func init() {
 
 		// only set the retryer on request if config doesn't have a retryer
 		if r.Config.Retryer == nil && (r.Operation.Name == opModifyNetworkInterfaceAttribute || r.Operation.Name == opAssignPrivateIpAddresses) {
+			maxRetries := client.DefaultRetryerMaxNumRetries
+			if m := r.Config.MaxRetries; m != nil && *m != aws.UseServiceDefaultRetries {
+				maxRetries = *m
+			}
 			r.Retryer = client.DefaultRetryer{
-				NumMaxRetries:    client.DefaultRetryerMaxNumRetries,
```

Comment:
Customizations tend to be variable in different SDKs. Can we sync up with other SDKs on this? 


Also I think there's a possibility this was intended to limit the number of max retries allowed for AssignPrivateIpAddress operation or ModifyNetworkInterfaceAttribute. How do we verify that's not the case here?

Reply:
Sure glad to. In this case the SDK was customized to use alternate retry delays for these two EC2 operations.  I think this behavior has been removed in v2's "updated" retry model. The SDK not looking at number of max retries was a miss i think. It incorrectly used the customer's configured number of max retries, and instead defaulted to the client's default by mistake.

Reply:
For background looks like the custom behavior was originally added in, https://github.com/aws/aws-sdk-go/pull/1787, and later when the `DefaultRetryer` had configuration options added for retry delays, the usage of the configured max retries was  lost in translation, https://github.com/aws/aws-sdk-go/pull/2830.